core/source/api/opencmiss/zinc/scene.h

```diff
@@ -398,6 +415,102 @@ ZINC_API int cmzn_scene_get_spectrum_data_range(cmzn_scene_id scene,
 	int valuesCount, double *minimumValuesOut, double *maximumValuesOut);
 
 /**
+ * Clear transformation field or matrix in scene.
+ *
+ * @param scene  The scene to modify.
+ * @return  Result OK on success, ERROR_ARGUMENT on failure.
+ */
+ZINC_API int cmzn_scene_clear_transformation(cmzn_scene_id scene);
+
+/**
+ * Query whether scene has either a transformation field or matrix active.
+ *
+ * @param scene  The scene to query.
+ * @return  True scene has transformation, false if not or invalid scene.
+ */
+ZINC_API bool cmzn_scene_has_transformation(cmzn_scene_id scene);
+
+/**
+ * Get optional field providing the 4x4 transformation of the scene local
+ * coordinates into the parent scene local coordinate system, or world
+ * coordinate system if scene is for the root region.
+ * @see cmzn_scene_set_transformation_field
+ *
+ * @param scene  The scene to query.
+ * @return  Handle to field, or NULL/invalid handle if none or failed.
+ */
+ZINC_API cmzn_field_id cmzn_scene_get_transformation_field(cmzn_scene_id scene);
+
+/**
+ * Set optional field providing the 4x4 transformation of the scene local
+ * coordinates into the parent scene local coordinate system, or world
+ * coordinate system if scene is for the root region.
+ * Overrides any current scene transformation matrix.
+ * @see cmzn_scene_set_transformation_matrix
+ *
+ * @param scene  The scene to modify.
+ * @param transformation_field  A 16 component field whose values give a 4x4
+ * transformation of homogeneous coordinates, or NULL/invalid to clear. This
+ * field must be spatially constant, but it may be time-varying in which case
+ * scene redraws are automatically trigger when the time changes. Runtime
+ * errors will be reported if this field cannot be evaluated given only the
+ * current timekeeper time. Field must belong to this scene's owning region.
+ * Alternatively, iff a transformation field is in use, passing an invalid/NULL
```

Comment:
No, it returns OK.
Since writing this I added cmzn_scene_clear_transformation/Scene::clearTransformation, so passing a null field to the above function is now just an extra way to cancel transformation, and I agree it's weak to say it is only effective if a transformation field is currently in use.
I think the way to go is either:
(1) insist that a field is passed to the above function and require the use of clearTransformation to cancel, OR
(2) make passing a null field into the above function clear transformation always. Sure it can return an error if there was no current transformation.
Do you have a preference?